### DISCUSSION TITLE
custom the length of the subdomain payload, for length-limited input scenarios.

### DISCUSSION THREAD

**Discussion by webca7** [projectdiscovery/interactsh#118] [Q&A] [answered]
custom the length of the subdomain payload, for length-limited input scenarios.

The server adds an allowable length range parameter, example: -payload-length 1-16
The client specifies the length of the generated payload, example: -payload-length 4

**Comment by kmahyyg** [comment]
same problem here, for self-hosted version, maybe you could allow us to custom maximum subdomain length, like 6 to 10 for self-hosted instances.

**Comment by ehsandeep** [comment, marked as answer]
@kmahyyg @webca7 interactsh server supports global/wildcard interaction (`-root-tld`) with self-hosted server and that might be a better option to use instead of using of custom/short length payload which is not supported to avoid payload collisions.

**Reply by kmahyyg** [reply]
Imagine even you have a self-hosted instance, there's still a lot of different users to use this. Since Chinese-owned `dnslog.cn` only use 6 char as second-level prefix, it still works well, they have about 1B users. We don't really need such long prefix since only `[A-Z][a-z][0-9]{6}` has about `(26+26+10)^6=568,0023,5584` possibilities.  

As for me, I use `<random>.second-level.interactsh.com` per-time, it still works fine. `62^33=140904788838798583412024976944907738550096881878156624003072` , too long and too large.
